### DISCUSSION TITLE
Scenario Analysis

### DISCUSSION THREAD

**Discussion by shubhambooking2** [google/meridian#197] [General]
Is it possible to run scenario analysis in the sense that let's say i already have a model trained at daily level and i have a new dataset that contains the projected spend and associated impressions/clicks at the daily level for all media channels for next 100 days. Given this distribution of spend and associated clicks/impressions, i would like to understand what would be the corresponding ROI if we spent across the media channels as provided(data for next 100 days)

**Comment by jyklly** [comment]
At the moment this is not directly supported. The `roi` function in `Analyzer` (https://developers.google.com/meridian/reference/api/meridian/analysis/analyzer/Analyzer?hl=en#roi) allows for `new_media` and `new_media_spend` but requires the dimensions of these inputs to be the same as the training data.

We are looking into directly supporting future media but in the meantime, there is a workaround you can maybe try out:

In the call to the `roi` function `new_media` has to have the same time granularity as the training data, and currently it must also have the same number of time periods. A workaround, though, is to set `new_media` to a tensor containing the future media values for the initial 100 days, and zero for all time periods after. Similarly, `new_media_spend` should contain the spend associated with the initial 100 days and zero after.
